test/lib/jdk/test/lib/hotspot/ir_framework/TestFramework.java

```diff
@@ -0,0 +1,1044 @@
+     */
+    public static final Set<String> JTREG_WHITELIST_FLAGS = new HashSet<>(
+            Arrays.asList(
+                    // The following substrings are part of more than one VM flag
+                    "RAM",
+                    "Heap",
+                    "Trace",
+                    "Print",
+                    "Verify",
+                    "TLAB",
+                    "UseNewCode",
+                    // The following substrings are only part of one VM flag (=exact match)
+                    "CreateCoredumpOnCrash",
+                    "IgnoreUnrecognizedVMOptions",
+                    "UnlockDiagnosticVMOptions",
+                    "UnlockExperimentalVMOptions",
+                    "BackgroundCompilation",
+                    "Xbatch",
+                    "TieredCompilation"
+            )
+    );
+
+    static final boolean VERBOSE = Boolean.getBoolean("Verbose");
+    static final boolean TESTLIST = !System.getProperty("Test", "").isEmpty();
+    static final boolean EXCLUDELIST = !System.getProperty("Exclude", "").isEmpty();
+    static final String TEST_VM_FLAGS_START = "##### TestFrameworkPrepareFlags - used by TestFramework #####";
+    static final String TEST_VM_FLAGS_DELIMITER = " ";
+    static final String TEST_VM_FLAGS_END = "----- END -----";
+    static final String RERUN_HINT = """
+                                       #############################################################
+                                        - To only run the failed tests use -DTest, -DExclude,
+                                          and/or -DScenarios.
+                                        - To also get the standard output of the test VM run with\s
+                                          -DReportStdout=true or for even more fine-grained logging
+                                          use -DVerbose=true.
+                                       #############################################################
+                                     """ + System.lineSeparator();
+
+    private static final int WARMUP_ITERATIONS = Integer.getInteger("Warmup", -1);
+    private static final boolean PREFER_COMMAND_LINE_FLAGS = Boolean.getBoolean("PreferCommandLineFlags");
+    private static final boolean EXCLUDE_RANDOM = Boolean.getBoolean("ExcludeRandom");
+    private static final boolean REPORT_STDOUT = Boolean.getBoolean("ReportStdout");
+    private final boolean VERIFY_VM = Boolean.getBoolean("VerifyVM") && Platform.isDebugBuild();
+    private boolean VERIFY_IR = Boolean.parseBoolean(System.getProperty("VerifyIR", "true"));
+
+    private boolean shouldVerifyIR; // Should we perform IR matching?
+    private static String lastTestVMOutput;
+    private static boolean toggleBool;
+
+    private final Class<?> testClass;
+    private Set<Class<?>> helperClasses = null;
+    private List<Scenario> scenarios = null;
+    private Set<Integer> scenarioIndices = null;
+    private List<String> flags = null;
```

Comment:
`null` is a default value for reference types, so there is no need to set it explicitly.